### ISSUE TITLE
Old 0.1 version on the latest docs

### ISSUE THREAD

**Issue by atugushev** [jazzband/pip-tools#1939]
RTD built [latest](https://pip-tools.readthedocs.io/en/latest/) doc on b55abdc31e5a905389da26352a9beab2a6c87bfa commit using `0.1.dev128` version.

#### Expected result

There should be `v7.1.1.devN`

#### Actual result

Screenshot:
<img width="1130" alt="image" src="https://github.com/jazzband/pip-tools/assets/7377671/1c9bd68c-9c63-452b-a6da-31274ced6643">

#### Additional context

Logs on [RTD](https://readthedocs.org/projects/pip-tools/builds/21423718/):
```
$ python -m sphinx -T -E -W --keep-going -b dirhtml ...
Running Sphinx v6.2.1
pip-tools version: 0.1.dev128
pip-tools release: 0.1.dev128
...
```

**Comment by webknjaz** [comment]
I think this happens because they make a shallow clone. I like to unshallow it to make the machinery discover the proper version.

**Comment by chrysle** [comment]
Running the `tox` `build-docs` env succeeds detecting the right version:

```
pip-tools version: 7.1.1
pip-tools release: 7.1.1.dev5
```
There is run `git fetch --unshallow`.

We might be able to unshallow the clone in the RTD configuration like this:

```yaml
build:
  os: ubuntu-22.04
  jobs:
    post_checkout:
      - git fetch --unshallow  || true
```

**Comment by atugushev** [comment]
@chrysle thanks for the snippet. Addressed in #1941.
